src/Transports/MassTransit.KafkaIntegration/KafkaIntegration/KafkaBaseReceiveContext.cs

```diff
@@ -14,31 +13,30 @@ public sealed class KafkaReceiveContext<TKey, TValue> :
         ReceiveLockContext
         where TValue : class
     {
-        readonly IHeadersDeserializer _headersDeserializer;
-        readonly IConsumerLockContext<TKey, TValue> _lockContext;
-        readonly ConsumeResult<TKey, TValue> _result;
+        readonly ConsumerContext<TKey, TValue> _consumerContext;
+        readonly ConsumeResult<byte[], byte[]> _result;
         IHeaderProvider _headerProvider;
+        TKey _key;
 
-        public KafkaReceiveContext(ConsumeResult<TKey, TValue> result, ReceiveEndpointContext receiveEndpointContext,
-            IConsumerLockContext<TKey, TValue> lockContext, IHeadersDeserializer headersDeserializer)
+        public KafkaReceiveContext(ConsumeResult<byte[], byte[]> result, ReceiveEndpointContext receiveEndpointContext,
```

Comment:
Like KafkaReceiveEndpointContext might be the ticket here, with all those things on it (key, value serializer, etc., etc.).